УРААААААААААААААА, ГИТ РАБОТАЕТ, СПАСИБО.

STGPSYCHO opened the pull request:
Сделана 2 лаба
@ysn-a

STGPSYCHO committed:
УРААААААААААААААА, ГИТ РАБОТАЕТ, СПАСИБО.

Сделана 2 лаба

ysv-a commented on src/scss/blocks/sign-in-button.scss:
```diff
@@ -0,0 +1,27 @@
+    width: 32px;
+    height: 32px;
+}
+.sign-in-button__item-name{
+    font-weight: 400;
+    font-size: 15px;
+    color: #F5F5F5;
+    @include media-xl-max{
+        display: none;
+    }
+    @include media-lg-max{
+        display: none;
+    }
+    @include media-md-max{
+        display: none;
+    }
+    @include media-sm-max{
+        display: none;
+    }
+}
```
логику с display: none; нужно подправить, зачем на все ендпоинты один и тот же css

STGPSYCHO replied:
@ysv-a  А как можно скрыть этот элемент во всех состояниях экрана меньше чем medai-xl-max? Я вроде по лекциям искал, не нашел подобного. Вот такую реализацию только замутил

ysv-a replied:
это значит что если экран будет меньше 1260px, то применить правила
```suggestion
.sign-in-button__item-name{
    font-weight: 400;
    font-size: 15px;
    color: #F5F5F5;
    @include media-xl-max{
        display: none;
    }
}
```

STGPSYCHO replied:
готово

ysv-a commented on src/scss/blocks/navigation.scss:
```diff
@@ -0,0 +1,81 @@
+    overflow: hidden;
+}
+.navigation__item:hover{
+    /* height: 36px; */
```
все комментарии старайтесь убирать перед ревью

STGPSYCHO replied:
готово

ysv-a commented on src/js/main.js:
```diff
@@ -0,0 +1,152 @@
+const filmButton = document.querySelector('#film-name-button');
+function handlerProperty(e){
+  console.log(e)
+  const films = ['Алиса в стране чудес', 'Мстители','Город мертвых', 'Призрачный гонщик']
+
+
+  // Посмотреть innerHTML. и обнулить контейнер
+  const filmsList = document.querySelector('.usefull-info-left__list');
+  filmsList.innerHTML = ''
+  // while (filmsList.firstChild) {                       Было / Стало
+  //   filmsList.removeChild(filmsList.firstChild);
+  // }
+  
+  
+  let input = document.getElementById('film-name-text').value ;
+  
+  
+  const filterFilmsList = films.filter( (film) => film.includes(input));
+  
+  const contentString =
+  filterFilmsList.map((item,index) => `
+      <li class="usefull-info-left__item">${item}</li>
+    `).join('') 
+  filmsList.insertAdjacentHTML('beforeend', contentString);
+}
+if(filmButton){
+  filmButton.addEventListener('click', handlerProperty )  
+}
+handlerProperty('click');
```
тут конечно не все корректно, нужно за пределы функции вынести константы, при нажатии на кнопку, заново создается массив и ищется filmsList 

и нужно не забывать убирать комментарии

```suggestion
const filmButton = document.querySelector("#film-name-button");
const films = [
  "Алиса в стране чудес",
  "Мстители",
  "Город мертвых",
  "Призрачный гонщик",
];
const filmsList = document.querySelector(".usefull-info-left__list");
function handlerProperty() {
  let input = document.getElementById("film-name-text").value;
  filmsList.innerHTML = "";

  const filterFilmsList = films.filter((film) => film.includes(input));

  const contentString = filterFilmsList
    .map(
      (item, index) => `
      <li class="usefull-info-left__item">${item}</li>
    `
    )
    .join("");
  filmsList.innerHTML = contentString;
}
if (filmButton) {
  filmButton.addEventListener("click", handlerProperty);
}
handlerProperty();
```

ysv-a replied:
ну и insertAdjacentHTML тоже тут не нужен, мы же контейнер очищаем, он пустой, поэтому можем просто через innerHTML вставить новые узлы

ysv-a commented on src/js/main.js:
```diff
@@ -0,0 +1,152 @@
+            <p class="article__text">${input_text}</p>    
+        </div>
+        <footer>
+            <p class="article__date">${now.toLocaleString('en', {month:'long'})} ${now.getDate()}, ${now.getFullYear()} · 4 min read</p>
```
проще вынести в отдельную переменную
${now.toLocaleString('en', {month:'long'})} ${now.getDate()}, ${now.getFullYear()}

STGPSYCHO replied:
готово

ysv-a commented on src/js/main.js:
```diff
@@ -0,0 +1,152 @@
+  });
+}
+
+const faq = document.getElementsByClassName("faq-page__question");
```
почему не querySelector?

ysv-a commented on src/js/main.js:
```diff
@@ -0,0 +1,152 @@
+for (i = 0; i < faq.length; i++) {
+    faq[i].addEventListener("click", function () {
+
+        this.classList.toggle("active");
```
active, как бы ок. но лучше по бему сделать, faq-page__question--active

STGPSYCHO replied:
Если честно уже не нашел в своем коде класс active.. решил, что не стоит ломать то, что работает. Дальше буду обращать внимание

ysv-a commented on src/html/index.html:
```diff
@@ -0,0 +1,192 @@
+            <div class="faq container" id="faq">
+                <h3 class="faq-title">FAQ</h2>
+                <dl class="faq-page">
+                    <dt class="faq-page__question">Use service data to identify your most common questions ?</dt>
+                    <dd class="faq-page__answer">Your FAQ page should address the most common questions customers have about your products, services, and brand as a whole. The best way to identify those questions is to tap into your customer service data and see which problems customers are consistently reaching out to you with. If you have a help desk or call center software you can easily disco ...
+        
+                    <dt class="faq-page__question">Include space for live support options</dt>
+                    <dd class="faq-page__answer">FAQ pages are intended as an initial support option for customers looking for an immediate answer to a quick question or problem. It shouldn't replace your knowledge base or your entire support team, but rather supplement your support channels as an additional, lightweight resource. When customers do have questions that extend beyond the range of y ...
+        
+                    <dt class="faq-page__question">Update content and add more solutions over time.</dt>
+                    <dd class="faq-page__answer">As your business grows, you should update your FAQ page to include any new questions that become popular over time. For example, if you release a new product or service you'll need to update your FAQ page so that it includes troubleshooting tips for this offer as well. This shows that you take customer success seriously and that you're rolling out  ...
```
в dt можно вставить кнопку, а в кнопку например span или svg иконку
`<dt><button>>Use service data to identify your most common questions ? <svg></svg></button></dt>`

ysv-a replied:
тут кнопку и иконку лучше добавить все таки

ysv-a commented on src/html/index.html:
```diff
@@ -0,0 +1,192 @@
+                </article>
+            </div>
+
+            <div class="article-add">
+                <div class="article-add__button">
+                    <button class="article-add__button-add" id="article-add__button">Добавить</button>
+                </div>
+                <form class="article-add__form form" onsubmit="return false;">
+                    <div class="form__inputs">
+                        <label for="article-text__name" class="article-text-name">Название статьи</label>
+                        <input type="text" id="article-text-name">
+                        <label for="article-text__url" class="article-text-url">Ссылка на картинку</label>
+                        <input type="text" id="article-text-url">
+                        <label for="article-text-description" class="article-text__description">Описание</label>
+                        <textarea  id="article-text-description"></textarea>
+                        <div class="form__buttons">
+                            <button type="reset" class="article-add__button-cancel" id="article-add__button-cancel">Отмена</button>
+                            <button type="submit" class="article-add__button-create" id="article-add__button-create">Создать</button>
+                        </div>
+                    </div>
+                </form>
+            </div>
```
при нажатии на добавить блог поста пропадает блок FAQ

после создание блог поста, форму нужно сбрасывать

если выбран показ постов в строчку (list) и добавить блог пост, то он отображается не корректно

https://i.gyazo.com/d8876e9cbd642fde57b6c4b2e14bc91d.png

STGPSYCHO replied:
> при нажатии на добавить блог поста пропадает блок FAQ
@ysv-a  Сделал так, потому что в фигме такая логика. Если не нужно, то уберу

ysv-a replied:
да, сделать как описал я,
самое главное поправить логику работы переключения постов, чтоб не было как на скрине

STGPSYCHO replied:
готово

ysv-a reviewed (commented):
id в верстке лучше не исп, но раз сделали, то пусть остается

id можно исп. только для js, или выделить спец. классы, например .js-article-text-name
но css лучше не накатывать на такие классы и идентификаторы

#article-text-name
#article-text-url
#article-text-description

---

STGPSYCHO committed:
Исправление комментариев по PR

STGPSYCHO commented:
@ysv-a , все комментарии исправил.

ysv-a commented on src/html/index.html:
```diff
@@ -0,0 +1,192 @@
+                    </div>
+                </form>
+            </div>
+            <div class="faq container" id="faq">
```
` class="faq container" ` тут класс container не нужен, мы же и так в нем находимся уже

в моб. версии сделать вн. отступы поменьше, ну и размер шрифта, и справа отступа нет
(например по 15px)

https://i.gyazo.com/ce4c27918d81bd316f720e492074e461.png

STGPSYCHO replied:
Поправил.

ysv-a commented on src/scss/blocks/faq.scss:
```diff
@@ -0,0 +1,42 @@
+        padding-left: 32px;
+        margin-bottom: 30px;
+            
+            &::after{
+                width: 0;
+                height: 0;
+                border-style: solid;
+                border-width: 10px 10px 0 10px;
+                border-color: #000000 transparent transparent transparent;
+            }
+        }
```
вот эта логика я как понимаю сейчас не работает?

STGPSYCHO replied:
Да. Забыл удалить.

ysv-a reviewed (commented):
и поправить адаптив, там где добавить новые блог пост и кнопка и форма
https://i.gyazo.com/3e6fad739954f3f88eaaec38db86c1b0.png

видимо ширина фикс. стоит
фикс лучше не делать, обычно max-width

STGPSYCHO committed:
Адаптив на форму. Исправления по faq

STGPSYCHO commented:
@ysv-a , можно проверять последние замечания.

ysv-a commented on src/scss/blocks/navigation.scss:
```diff
@@ -0,0 +1,73 @@
+    flex-direction: row;
+    align-items: center;
+    padding: 0px;
+    @include media-md-max{
+        display: none;
+        gap: 0px;
+    }
+    @include media-sm-max{
+        display: none;
+        gap: 0px;
+    }
```
убрать дублирования css в медиа запросах

ysv-a commented on src/scss/blocks/article.scss:
```diff
@@ -0,0 +1,173 @@
+    font-weight: 400;
+    font-size: 14px;
+    color: #6B7280;
+    @include media-md-max{
+        font-size: 12px;
+    }
+    @include media-sm-max{
+        font-size: 12px;
+    }
```
убрать дублирования css в медиа запросах

ysv-a commented on src/scss/blocks/article.scss:
```diff
@@ -0,0 +1,173 @@
+    font-size: 16px;
+    color: #6B7280;
+
+    @include media-md-max{
+        font-size: 14px;
+    }
+    @include media-sm-max{
+        font-size: 14px;
+    }
```
убрать дублирования css в медиа запросах

ysv-a commented on src/scss/blocks/usefull-info.scss:
```diff
@@ -0,0 +1,186 @@
+    height: auto;
+
+
+     @include media-md-max {
+        grid-template-columns: repeat(1, minmax(0, 1fr));
+        grid-row-gap: 10px;
+        grid-column-gap: 0px;
+     }
+     @include media-sm-max {
+        grid-template-columns: repeat(1, minmax(0, 1fr));
+        grid-row-gap: 10px;
+        grid-column-gap: 0px;
+    }
```
убрать дублирования css в медиа запросах

ysv-a commented on src/scss/blocks/usefull-info.scss:
```diff
@@ -0,0 +1,186 @@
+    list-style-type:disc;
+    font-weight: 400;
+    font-size: 16px;
+    @include media-lg-max{
+        font-size: 14px;
+    }
+    @include media-md-max{
+        font-size: 14px;
+    }
+    @include media-sm-max{
+        font-size: 14px;
+    }
```
убрать дублирования css в медиа запросах

ysv-a reviewed (commented):
1. в моб версии кнопка выходит за границы
https://i.gyazo.com/eb21f15a376e5d874920e4141aed49a6.png

2. в десктопной версии, форма стала по всю ширину
https://i.gyazo.com/9800e1c4255f2fe12deb74b641d53633.png

3. в моб версии почему то исчезли переключатели контента
https://i.gyazo.com/5e3c5963d881713cc37d0f05b9619ac2.png

4. убрать дублирования css в медиа запросах 
не везде отметил, но много где дублируется код

`.article__text{
    font-weight: 400;
    font-size: 16px;
    color: #6B7280;

    @include media-md-max{
        font-size: 14px;
    }
    @include media-sm-max{
        font-size: 14px;
    }
}`
в точке media-md-max, определив 14px, значит все что ниже media-md-max также будет 14px

нет смысла в 
`@include media-sm-max{
        font-size: 14px;
    }`

STGPSYCHO committed:
1-3 пункта PR

в моб версии кнопка выходит за границы
https://i.gyazo.com/eb21f15a376e5d874920e4141aed49a6.png

в десктопной версии, форма стала по всю ширину
https://i.gyazo.com/9800e1c4255f2fe12deb74b641d53633.png

в моб версии почему то исчезли переключатели контента
https://i.gyazo.com/5e3c5963d881713cc37d0f05b9619ac2.png

STGPSYCHO committed:
4 пункт PR

убрать дублирования css в медиа запросах
не везде отметил, но много где дублируется код

STGPSYCHO commented:
@ysv-a можно проверять.

> в моб версии кнопка выходит за границы
https://i.gyazo.com/eb21f15a376e5d874920e4141aed49a6.png
>
пофиксил



>в десктопной версии, форма стала по всю ширину
 https://i.gyazo.com/9800e1c4255f2fe12deb74b641d53633.png
>
Пофиксил

> в моб версии почему то исчезли переключатели контента
https://i.gyazo.com/5e3c5963d881713cc37d0f05b9619ac2.png
> 
Сделано специально, так как отображение листом на такую маленькую ширину не имеет смылса.
Даже если в большом разрешении выбрано отображение листом, я принудительно отрисовываю в столбец.

> убрать дублирования css в медиа запросах
не везде отметил, но много где дублируется код
> 
когда делал не знал такую штуку. Почему то казалось что явно нужно указывать каждый раз. Исправил

ysv-a commented on src/scss/blocks/navigation.scss:
```diff
@@ -0,0 +1,69 @@
+    width: 252px;
+    height: 60px;
+    margin-right: 30px;
+    @include media-md-max{
+        display: none;
+    }
+    @include media-sm-max{
+        display: none;
+    }
```
в общем осталось убрать дублирования и все, лучше еще раз внимательно пробежаться по стилям и пофиксить

STGPSYCHO committed:
Так, ну вроде точно удалил все дубли

STGPSYCHO commented:
@ysv-a , удалил везде дубли, пробежался по коду. 
Вроде даже ничего не сломал

ysv-a reviewed (commented):
можно мерджить
